lib/specinfra/ext/class.rb

```diff
@@ -2,7 +2,7 @@ class Class
   def subclasses
     result = []
     ObjectSpace.each_object(Class) do |k|
-      next if k.name.nil?
+      next unless self == ancestors.first #Skips if this is a singleton class
```

Comment:
That's a decent way to check but we want to check if `k` is a singleton class so under this version it would be:

```ruby
next unless k == k.ancestors.first # Skips if k is a singleton class
```